File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = defaultdict(list)
        result = []

        for key in strs:
            sorted_key = "".join(sorted(key))
            anagrams[sorted_key].append(key)

        for k, v in anagrams.items():
            result.append(v)

        return result

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> hm = new HashMap<>();

        for (String st: strs) {
            char[] charr = st.toCharArray();
            Arrays.sort(charr);
            String sortedKey = new String(charr);

            if (!hm.containsKey(sortedKey)) {
                hm.put(sortedKey, new ArrayList<>());
            }
            hm.get(sortedKey).add(st);
        }

        return new ArrayList<>(hm.values());
    }
}